.github/workflows/ci.yml

```
@@ -1,45 +1,111 @@
-name: Node CI
+name: CI
 
 on:
   push:
     branches:
       - main
-    tags:
-      - 'v*'
-  pull_request:
+      - master
+  pull_request: {}
+
+concurrency:
+  group: ci-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
 
 jobs:
-  lint:
+  test:
+    name: "Tests"
     runs-on: ubuntu-latest
-    outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    timeout-minutes: 10
+
     steps:
       - uses: actions/checkout@v4
       - uses: wyvox/action-setup-pnpm@v3
-      - run: pnpm lint
+      - name: Lint
+        run: pnpm lint
+      - name: Run Tests
+        run: pnpm test
 
-  test:
+  floating:
+    name: "Floating Dependencies"
     runs-on: ubuntu-latest
-    outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    timeout-minutes: 10
+
     steps:
       - uses: actions/checkout@v4
       - uses: wyvox/action-setup-pnpm@v3
```

```suggestion
      - uses: wyvox/action-setup-pnpm@v3
        with:
         args: "--no-lockfile"
```
installation happens for you out of the box

---

https://github.com/wyvox/action-setup-pnpm?tab=readme-ov-file#args

---

done